### DISCUSSION TITLE
issue: Nested field arrays not inheriting type

### DISCUSSION THREAD

**Discussion by Niesyto** [react-hook-form/react-hook-form#7991] [Q&A]
### Version Number

7.27.1

### Codesandbox/Expo snack

https://codesandbox.io/s/rqtjib

### Steps to reproduce

1. Go to codesanbox (that's all, I've provided defaultValues which should load everything)

### Expected behaviour

I would like to see nested array values. Presumably, they are not displayed because `NestedArray` cannot properly inherit type on `fields` and instead sees it as `Record<"id", string>[]`

The example I've provided is a simplified version of my issue (i have one more level of nesting). 

I've tried passing `control` instead of using `useFormContext`, but it doesn't help

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please see this issue for a similar report: https://github.com/react-hook-form/react-hook-form/issues/7972 and we are improving that on v8.

**Comment by bluebill1049** [comment]
also we don't support flat array as well.

https://react-hook-form.com/api/usefieldarray

**Reply by bluebill1049** [reply]
some more issue in the codesandbox, i have fix a few: https://codesandbox.io/s/ancient-voice-8u3ccs?file=/src/App.tsx
